firefox-ios/Client/Frontend/Browser/MainMenu/MainMenuConfigurationUtility.swift

```diff
@@ -0,0 +1,297 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Foundation
+import MenuKit
+import Shared
+
+struct MainMenuConfigurationUtility: Equatable {
+    func generateMenuElements(
+        with uuid: WindowUUID,
+        tabIsHomepage isHomepage: Bool?
+    ) -> [MenuSection] {
+        // Always include these sections
+        var menuSections: [MenuSection] = [
+            getNewTabSection(with: uuid),
+            getLibrariesSection(with: uuid),
+            getOtherToolsSection(with: uuid, isHomepage: isHomepage ?? false)
+        ]
+
+        // Conditionally add tools section if this is a website
+        if let isHomepage = isHomepage, !isHomepage {
+            menuSections.insert(getToolsSection(with: uuid), at: 1)
+        }
+
+        return menuSections
+    }
+
+    private func getNewTabSection(with uuid: WindowUUID) -> MenuSection {
+        return MenuSection(options: [
+            MenuElement(
+                title: .MainMenu.TabsSection.NewTab,
+                iconName: "",
+                isEnabled: true,
+                isActive: false,
```

Comment:
I see that these two properties: `isEnabled` & `isActive` have often the same values. Wouldn't be helpful to have a convenience init with some default values? Just a thought, nothing more.

Reply:
These values are just placeholders for now. Once certain logic is in place, these will all be replaced by functions that result a true/false in them.